frontend/public/components/markdown-view.tsx

```diff
@@ -1,16 +1,63 @@
 import * as React from 'react';
 import * as _ from 'lodash-es';
-import { Converter } from 'showdown';
+import { Converter, extension, subParser } from 'showdown';
 import * as sanitizeHtml from 'sanitize-html';
+import { Spotlight } from '@console/shared';
 
 const tableTags = ['table', 'thead', 'tbody', 'tr', 'th', 'td'];
 
+extension('quickstart', () => {
```

Comment:
Okay, send this as a DM, but I figured I'd post the same here:

Make a new component for this extension and the needed execution code:
```
import * as React from 'react';
import { extension } from 'showdown';
import { Spotlight } from '../spotlight';
extension('quickstart', () => {
  return [
    {
      type: 'lang',
      regex: /\[(\w+)]{{(highlight)\s([\w-]+)}}/g,
      replace: (text: string, linkLabel: string, linkType: string, linkId: string): string => {
        return `<button class="pf-c-button pf-m-inline pf-m-link" data-highlight="${linkId}">${linkLabel}</button>`;
      },
    },
  ];
});
type MarkdownHighlightExtensionProps = {
  docContext: HTMLDocument;
};
const MarkdownHighlightExtension: React.FC<MarkdownHighlightExtensionProps> = ({ docContext }) => {
  const [selector, setSelector] = React.useState<string>(null);
  React.useEffect(() => {
    // console.debug('highlight nodes', this.frame.contentDocument.querySelectorAll('a[data-highlight]'));
    const elements = docContext.querySelectorAll('[data-highlight]');
    function startHighlight(e) {
      const highlightId = e.target.getAttribute('data-highlight');
      if (!highlightId) {
        return;
      }
      setSelector(null);
      setTimeout(() => {
        setSelector(`[data-tour-id="${highlightId}"]`);
      }, 0);
    }
    elements.forEach((elm) => elm.addEventListener('click', startHighlight));
    return () => {
      elements.forEach((elm) => elm.removeEventListener('click', startHighlight));
    };
  }, [docContext]);
  if (!selector) {
    return null;
  }
  return <Spotlight selector={selector} animate />;
};
export default MarkdownHighlightExtension;
```

Reply:
You'll also need to make other changes in this file:

```
diff --git frontend/public/components/markdown-view.tsx frontend/public/components/markdown-view.tsx
index 33a2c341e..3032ca74d 100644
--- frontend/public/components/markdown-view.tsx
+++ frontend/public/components/markdown-view.tsx
@@ -80,10 +35,12 @@ const markdownConvert = (markdown) => {
       'li',
       'code',
       'pre',
+      'button',
       ...tableTags,
     ],
     allowedAttributes: {
-      a: ['href', 'target', 'rel'],
+      a: ['href', 'target', 'rel', 'data-*'],
+      button: ['class', 'data-*'],
     },
     allowedSchemes: ['http', 'https', 'mailto'],
     transformTags: {
@@ -177,14 +134,20 @@ export class SyncMarkdownView extends React.Component<
       <body class="pf-m-redhat-font"><div style="overflow-y: auto;">${markdownConvert(
         content || 'Not available',
       )}</div></body>`;
+
     return (
-      <iframe
-        sandbox="allow-popups allow-same-origin"
-        srcDoc={contents}
-        style={{ border: '0px', display: 'block', width: '100%', height: '0' }}
-        ref={(r) => (this.frame = r)}
-        onLoad={() => this.updateDimensions()}
-      />
+      <>
+        <iframe
+          sandbox="allow-popups allow-same-origin"
+          srcDoc={contents}
+          style={{ border: '0px', display: 'block', width: '100%', height: '0' }}
+          ref={(r) => (this.frame = r)}
+          onLoad={() => this.updateDimensions()}
+        />
+        {this.frame?.contentDocument && (
+          <MarkdownHighlightExtension docContext={this.frame.contentDocument} />
+        )}
+      </>
     );
   }
 }
```